controller/src/main/java/org/jboss/as/controller/operations/global/ReadResourceHandler.java

```diff
@@ -500,6 +505,9 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
                 ModelNode value = ar.response;
                 if (!value.has(FAILURE_DESCRIPTION)) {
                     sortedAttributes.put(entry.getKey(), value.get(RESULT));
+                    if (value.hasDefined(RESPONSE_HEADERS, WARNINGS)) {
```

Comment:
I think it is better if you add this validation inside of the private method. It is common to both points and that will protect the method to be invoked when the value has not defined the RESPONSE_HEADERS/WARNINGS